modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/security/permission/resource/CPDisplayLayoutModelResourcePermission.java

```diff
@@ -0,0 +1,170 @@
+import com.liferay.commerce.product.service.CPDisplayLayoutLocalService;
+import com.liferay.commerce.product.service.CommerceCatalogLocalService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.security.permission.resource.PortletResourcePermission;
+import com.liferay.portal.kernel.service.LayoutLocalService;
+import com.liferay.portal.kernel.service.permission.LayoutPermissionUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portlet.asset.service.permission.AssetCategoryPermission;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Danny Situ
+ */
+@Component(
+	property = "model.class.name=com.liferay.commerce.product.model.CPDisplayLayout",
+	service = ModelResourcePermission.class
+)
+public class CPDisplayLayoutModelResourcePermission
+	implements ModelResourcePermission<CPDisplayLayout> {
+
+	@Override
+	public void check(
+			PermissionChecker permissionChecker,
+			CPDisplayLayout cpDisplayLayout, String actionId)
+		throws PortalException {
+
+		if (!contains(permissionChecker, cpDisplayLayout, actionId)) {
+			throw new PrincipalException.MustHavePermission(
+				permissionChecker, CPDisplayLayout.class.getName(),
+				cpDisplayLayout.getCPDisplayLayoutId(), actionId);
+		}
+	}
+
+	@Override
+	public void check(
+			PermissionChecker permissionChecker, long cpDisplayLayoutId,
+			String actionId)
+		throws PortalException {
+
+		if (!contains(permissionChecker, cpDisplayLayoutId, actionId)) {
+			throw new PrincipalException.MustHavePermission(
+				permissionChecker, CPDisplayLayout.class.getName(),
+				cpDisplayLayoutId, actionId);
+		}
+	}
+
+	@Override
+	public boolean contains(
+			PermissionChecker permissionChecker,
+			CPDisplayLayout cpDisplayLayout, String actionId)
+		throws PortalException {
+
+		if (actionId.equals(ActionKeys.VIEW) &&
+			Validator.isNotNull(cpDisplayLayout.getLayoutUuid()) &&
+			!LayoutPermissionUtil.contains(
+				permissionChecker, _getLayout(cpDisplayLayout),
+				ActionKeys.VIEW)) {
+
+			return false;
+		}
+
+		String className = cpDisplayLayout.getClassName();
+
+		if (className.equals(CPDefinition.class.getName())) {
+			CPDefinition cpDefinition =
+				_cpDefinitionLocalService.getCPDefinition(
+					cpDisplayLayout.getClassPK());
+
+			CommerceCatalog commerceCatalog =
+				_commerceCatalogLocalService.fetchCommerceCatalogByGroupId(
+					cpDefinition.getGroupId());
+
+			if (!_commerceCatalogModelResourcePermission.contains(
+					permissionChecker, commerceCatalog, actionId)) {
```

Comment:
I don't think that for deleting the Page for the product the user should have DELETE permission, it should be enough the UPDATE, because here you are not actually deleting a product you are  modifying a relation with a related entity